File: server/app/models/feedback.py
"""
피드백 관련 데이터베이스 모델 (User 관계 제거)
"""
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.sql import func

from app.core.database import Base


class Feedback(Base):
    """사용자 피드백 모델 (auth 없이는 User 관계 없음)"""
    __tablename__ = "feedback"

    id = Column(String(50), primary_key=True, index=True)
    # user_id = Column(Integer, ForeignKey("users.id"), nullable=True)  # auth 없이는 사용 불가
    type = Column(String(20), nullable=False)  # bug, feature, improvement, other
    title = Column(String(200), nullable=False)
    content = Column(Text, nullable=False)
    rating = Column(Integer, nullable=True)  # 1-5
    contact_email = Column(String(255), nullable=True)  # 비회원용 연락처
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # 관계 정의 (auth 없이는 사용 불가)
    # user = relationship("User", back_populates="feedback")

File: mobile/lib/models/feedback.dart
class FeedbackType {
  static const String bug = 'BUG';
  static const String feature = 'FEATURE';
  static const String improvement = 'IMPROVEMENT';
  static const String other = 'OTHER';

  static String getDisplayName(String type) {
    switch (type) {
      case bug:
        return '버그 리포트';
      case feature:
        return '기능 제안';
      case improvement:
        return '개선 제안';
      case other:
        return '기타';
      default:
        return '기타';
    }
  }

  static List<Map<String, String>> getAll() {
    return [
      {'value': bug, 'label': '버그 리포트'},
      {'value': feature, 'label': '기능 제안'},
      {'value': improvement, 'label': '개선 제안'},
      {'value': other, 'label': '기타'},
    ];
  }
}

class Feedback {
  final int? id;
  final String feedbackType;
  final String title;
  final String content;
  final String? contactEmail;
  final DateTime? createdAt;

  Feedback({
    this.id,
    required this.feedbackType,
    required this.title,
    required this.content,
    this.contactEmail,
    this.createdAt,
  });

  factory Feedback.fromJson(Map<String, dynamic> json) {
    return Feedback(
      id: json['id'] as int?,
      feedbackType: json['feedback_type'] as String,
      title: json['title'] as String,
      content: json['content'] as String,
      contactEmail: json['contact_email'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'feedback_type': feedbackType,
      'title': title,
      'content': content,
      if (contactEmail != null && contactEmail!.isNotEmpty)
        'contact_email': contactEmail,
    };
  }
}
